sktime/covariance/online_covariance.py

```diff
@@ -100,45 +131,59 @@ def _create_model(self) -> OnlineCovarianceModel:
          Indices of those columns that are to be computed. If None, all columns are computed.
-        :param data:
-        :param weights:
+        :param data: list of sequences (n elements)
+        :param weights: list of weight arrays (n elements) or array (shape
         :param n_splits:
         :param column_selection:
         :return:
         """
+        # TODO: constistent dtype
+        data = ensure_timeseries_data(data)
+
         self._rc.clear()
+
         if n_splits is None:
-            if self.is_lagged:
-                dlen = len(data[0])
-            else:
-                dlen = len(data)
+            dlen = min(len(d) for d in data)
             n_splits = int(dlen // 100 if dlen >= 1e4 else 1)
-        if self.is_lagged:
-            x, x_lagged = data
-        else:
-            x, x_lagged = data, np.empty((0,))
-
-        x, x_lagged = np.asarray_chkfinite(x), np.asarray_chkfinite(x_lagged)
 
-        assert len(x_lagged) == 0 or len(x) == len(x_lagged), f"Expected data and time lagged data of equal length " \
-            f"but got {len(x)} != {len(x_lagged)}"
-
-        if weights is not None and len(np.atleast_1d(weights)) != len(x):
-            raise ValueError(f"Weights have incompatible shape "
-                             f"(#weights={len(weights) if weights is not None else None} != {len(x)}=#frames.")
-        wsplit = np.array_split(weights, n_splits) if weights is not None else [None] * n_splits
+        if lagtime is None:
+            lagtime = self.lagtime
+        else:
+            self.lagtime = lagtime
+        assert lagtime is not None
+
+        assert isinstance(data, list)
+        assert all(isinstance(e, np.ndarray) for e in data)
+        assert all(e.dtype in (np.float32, np.float64) for e in data)
+
+        lazy_weights = False
+        wsplit = itertools.repeat(None)
+
+        if weights is not None:
+            if hasattr(weights, 'weights'):
+                lazy_weights = True
+            elif len(np.atleast_1d(weights)) != len(data[0]):
+                raise ValueError(
+                    "Weights have incompatible shape "
+                    f"(#weights={len(weights) if weights is not None else None} != {len(data[0])}=#frames.")
+            elif isinstance(weights, np.ndarray):
+                wsplit = np.array_split(weights, n_splits)
 
-        for x_batch, y_batch, w in zip(np.array_split(x, n_splits), np.array_split(x_lagged, n_splits), wsplit):
-            assert len(x_batch) == len(y_batch) or (not self.is_lagged and len(y_batch) == 0)
-            assert w is None or len(x_batch) == len(w)
-            if not self.is_lagged:
-                self.partial_fit(x_batch, column_selection=column_selection, weights=w)
-            else:
-                self.partial_fit((x_batch, y_batch), column_selection=column_selection, weights=w)
+        if self.is_lagged:
+            for (x, y), w in zip(timeshifted_split(data, lagtime=lagtime, n_splits=n_splits), wsplit):
```

Comment:
i think this fails for `lazy_weights` - also can we implicitly convert everything to a lazy weight and internally only work with a function call?

Reply:
it does not fail, there is also a test for it in test_koopman_reweighting.
The wsplit variable is always iterable.